test/application/server-commands.test.js

```diff
@@ -0,0 +1,95 @@
+const simpleGit = require("simple-git");
+const git = simpleGit();
+
+const scenarioUpToDate = {
+    modified: [],
+    ahead: 0,
+    behind: 0,
+}
+
+describe('ServerCommands', () => {
+    const scenarioModifiedOutput = {
+        modified: [
+            "package-lock.json",
+            "package.json",
+            "server_src/lib/serverCommands.js",
+        ],
+        ahead: 0,
+        behind: 0,
+    };
+
+    jest.mock("child_process", () => {
+        return {
+            exec: () => Promise.resolve()
+        }
+    });
+
+    let wasReset = null;
+    let currentScenario = scenarioUpToDate;
+    jest.mock("simple-git", () => {
+        return () => {
+            return {
+                status: () => Promise.resolve(
+                    currentScenario
+                ),
+                pull: () => Promise.resolve(),
+                reset: (input) => {
+                    wasReset = true;
+                    return Promise.resolve();
+                },
+                checkIsRepo: () => true
+            }
+        }
+    });
+
+    jest.mock("../../server_src/utils/npm.utils", () => {
+        return () => {
+            return {
+                doWeHaveMissingPackages: () => Promise.resolve(true),
+                installMissingNpmDependencies: () => Promise.resolve(true),
+            }
+        };
+    });
+
+    const {SystemCommands} = require("../../server_src/lib/serverCommands");
+
+    it("should be able to see non-pushed commits", async () => {
+        const serverResponse = await SystemCommands.checkIfOctoFarmNeedsUpdatingAndUpdate( {},true);
+        expect(serverResponse.message).toBe("OctoFarm is already up to date! Your good to go!");
+        expect(serverResponse.haveWeSuccessfullyUpdatedOctoFarm).toBe(false);
+        expect(serverResponse.statusTypeForUser).toBe('success');
+    });
+
+    it("should be able to detect and fix uninstalled packages", async () => {
+    });
+
+    it("should be able to detect and fix uninstalled packages", async () => {
```

Comment:
- scenario for when we dont have missing packages, but unstaged changes **without** force
- scenario for when we dont have missing packages, but unstaged changes **with** force
- scenario for when git pull throws that git is not installed (just an idea?)
- scenario for when git does not connect (air-gapped?)
- scenario for when we dont have either (good to go)

More ideas?